presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkMutableRowPageInput.java

```diff
@@ -96,8 +96,16 @@ public Page getNextPage()
                 Iterator<Tuple2<MutablePartitionId, PrestoSparkMutableRow>> currentIterator = rowIterators.get(currentIteratorIndex);
                 if (currentIterator.hasNext()) {
                     PrestoSparkMutableRow row = currentIterator.next()._2;
-                    ByteBuffer buffer = row.getBuffer();
-                    output.writeBytes(buffer.array(), buffer.arrayOffset() + buffer.position(), buffer.remaining());
+                    if (row.getBuffer() != null) {
```

Comment:
curious: when the `PrestoSparkMutableRow` will be backed by `ByteBuffer` and when it will be backed by `byte[]`? -- maybe add a comment to explain it?

Reply:
I will be backed by `byte[]` only when deserialized with `public <T> T deserialize(byte[] array, int offset, int length, ClassTag<T> classTag)`

Reply:
Added more comments